docs/tutorials/creating-an-adapter.rst

```diff
@@ -135,40 +130,35 @@ parentheses form the capture group from which the argument is extracted.
 Using the Adapter
 -----------------
 
-In order to use the device we must first create a simulation configuration file, we
-shall create one named ``my_shutter_simulation.yaml``, and open it with our preferred
-editor. This file will be used to set up a simulation consisting of a `Source` named
-source which will produce a constant flux, the shutter which will act on the flux as
-per our implementation, and a `Sink` named sink which will recieve the resulting flux.
-The shutter will be given a ShutterAdapter which uses the default configuration for
-``host`` and ``port``. 
-
-.. code-block:: yaml
-
-    - tickit.core.components.device_simulation.DeviceSimulation:
-        adapters: []
-        device:
-          tickit.devices.source.Source:
-            value: 42.0
-        inputs: {}
-        name: source
-    - tickit.core.components.device_simulation.DeviceSimulation:
-        adapters:
-        - examples.devices.shutter.ShutterAdapter: {}
-        device:
-          examples.devices.shutter.Shutter:
-            default_position: 0.2
-            initial_position: 0.24
-        inputs:
-          flux: source:value
-        name: shutter
-    - tickit.core.components.device_simulation.DeviceSimulation:
-        adapters: []
-        device:
-          tickit.devices.sink.Sink: {}
-        inputs:
-          flux: shutter:flux
-        name: sink
+As explained in `Creating a Device`, an adapter allows us to to control a device
+externally. And in order to use the device we created a `ComponentConfig` for said
+device. This is defined in the same file as the device and adapter, and defines any
+default initial configuration values. As well as this, we overwrite the magic method
+`__call__()`, which return a `DeviceSimulation` object. In the `DeviceSimulation`
+object, we also define the device's adapter(s):
+
+.. code-block:: python
+
+    @dataclass
+    class Shutter(ComponentConfig):
+
+        default_position: float
+        initial_position: Optional[float] = None
+        host: str = "localhost"
+        port: int = 25565
+
+        def __call__(self) -> Component:
+            return DeviceSimulation(
+                name=self.name,
+                device=ShutterDevice(
+                    default_position=self.default_position,
+                    initial_position=self.initial_position,
+                ),
+                adapters=[ShutterAdapter(host=self.host, port=self.port)],
+            )
+
+As the adapter(s) are defined in the `ComponentConfig`, they are not required to be
+declared in the YAML for the simulation.
```

Comment:
I think this is slightly misleading as it implies you _can_ still do it the old way. I'm not actually sure if you can or can't but either way it's not supported. Maybe:

```suggestion
Using `ComponentConfig` the adapters are instantiated in Python so they are not referenced in the YAML config.
```